code/game/objects/effects/portals.dm

```diff
@@ -50,6 +50,17 @@
 	creator = null
 	return ..()
 
+/obj/effect/portal/proc/blockExiles(atom/movable/AM)

```

Comment:
make it run the check on mobs if they are in the contents of an object that passes through?